src/main/java/com/bakdata/streams_store/App.java

```diff
@@ -112,19 +98,34 @@ public static void main(String[] args) throws Exception {
         streams.start();
     }
 
+    private static StreamsBuilder buildIt(
+            StreamsBuilder builder,
+            AdminClient adminClient) throws ExecutionException, InterruptedException {
```

Comment:
Have you looked into whether this is really secure? I wouldn't suggest using an Admin Client to retrieve data from topics, since this API will be public facing.

Reply:
Changed to Consumer Client.